### PR TITLE
feat: [CODE-1057]: Added specified repository listing in resource groups 

### PR DESCRIPTION
**Pull request by rkapoor10**
### Summary
Added specified repository listing in resource groups in UI

#### Screenshots

Before - 
Screenshot 2023-11-03 at 10 21 37 PM

After - 

https://github.com/harness/harness-core-ui/assets/61417822/682ff2ec-07d3-4bbd-82fa-30762a6ff701

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  resourceScope,
+  resourceType,
+  onResourceSelectionChange
+}) => {
+  const { accountIdentifier, orgIdentifier, projectIdentifier } = resourceScope
+
+  const {
+    data: repositoriesListResult,
+    loading: repositoriesLoading
+    // error: errorFetchingRepositories,
```
Comment (cjlee01) [inline src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceRenderer.tsx:31]: remove commented lines if you dont need em

File: src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        data={repositoriesDataContent as ParsedColumnContent[]}
+        selectedData={selectedData}
+        columns={columns}
+        onSelectChange={onSelectChange}
+      />
+    </Container>
+  ) : (
+    <Layout.Vertical flex={{ align: 'center-center' }} spacing="small" className={css.noDataContainer}>
+      <Icon name="resources-icon" size={20} />
+      <Text font={{ variation: FontVariation.BODY1 }} color={Color.BLACK}>
```
Comment (cjlee01) [inline src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx:84]: nit: you can use icon={'resource-icon'} and iconProps={{size:20}} in the text component

Comment (rkapoor10) [reply src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx]: made the changes

File: src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceRenderer.tsx (hunk 1 above)
Comment (rkapoor10) [reply src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceRenderer.tsx]: removed

**Comment by rkapoor10** [comment]
retrigger prettier

**Comment by rkapoor10** [comment]
retrigger jest

**Comment by rkapoor10** [comment]
retrigger eslint

**Comment by rkapoor10** [comment]
> image image can you align the text and icon?

Screenshot 2023-11-10 at 4 59 41 PM

**Review by d1wilko — DISMISSED** [review]
LGTM

File: src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      <ResourceHandlerTable
+        data={repositoriesDataContent as ParsedColumnContent[]}
+        selectedData={selectedData}
+        columns={columns}
+        onSelectChange={onSelectChange}
+      />
+    </Container>
+  ) : (
+    <Layout.Vertical flex={{ align: 'center-center' }} spacing="small" className={css.noDataContainer}>
+      {/* <Icon name="resources-icon" size={20} /> */}
```
Comment (cjlee01) [inline src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx:82]: remove this comment if u dont need

Comment (rkapoor10) [reply src/modules/60-code/components/ResourceGroupModal/Repository/RepositoryResourceModalBody.tsx]: done

**Comment by rkapoor10** [comment]
retrigger eslint

**Review by cjlee01 — APPROVED** [review]

**Comment by rkapoor10** [comment]
retrigger eslint

**Review by hitesharinga — APPROVED** [review]

**Pull request merged**